token-metadata/program/src/processor/collection/unverify_collection.rs

```diff
@@ -1,71 +1,115 @@
-    let collection_mint = next_account_info(account_info_iter)?;
-    let collection_info = next_account_info(account_info_iter)?;
+    let collection_mint_info = next_account_info(account_info_iter)?;
+    let collection_metadata_info = next_account_info(account_info_iter)?;
     let edition_account_info = next_account_info(account_info_iter)?;
+
     let using_delegated_collection_authority = accounts.len() == 6;
 
-    assert_signer(collection_authority_info)?;
+    // Account validation.
     assert_owned_by(metadata_info, program_id)?;
-    assert_owned_by(collection_info, program_id)?;
-    assert_owned_by(collection_mint, &spl_token::id())?;
-    assert_owned_by(edition_account_info, program_id)?;
+    assert_signer(collection_authority_info)?;
+    assert_owned_by(collection_mint_info, &spl_token::id())?;
 
+    // Deserialize the collection item metadata.
     let mut metadata = Metadata::from_account_info(metadata_info)?;
-    let collection_data = Metadata::from_account_info(collection_info)?;
-
-    assert_collection_verify_is_valid(
-        &metadata.collection,
-        &collection_data,
-        collection_mint,
-        edition_account_info,
-    )?;
-    if using_delegated_collection_authority {
-        let collection_authority_record = next_account_info(account_info_iter)?;
-        assert_has_collection_authority(
-            collection_authority_info,
-            &collection_data,
-            collection_mint.key,
-            Some(collection_authority_record),
-        )?;
-    } else {
-        assert_has_collection_authority(
-            collection_authority_info,
-            &collection_data,
-            collection_mint.key,
-            None,
-        )?;
+
+    // First, if there's no collection set, we can just short-circuit
+    // since there's nothing to unverify.
+    let collection = match metadata.collection.as_mut() {
+        Some(collection) => collection,
+        None => return Ok(()),
+    };
+
+    // Short-circuit if it's already unverified.
+    if !collection.verified {
+        return Ok(());
     }
 
-    // This handler can only unverify non-sized NFTs
-    if collection_data.collection_details.is_some() {
-        return Err(MetadataError::SizedCollection.into());
+    // The collection parent must be the actual parent of the
+    // collection item.
+    if collection.key != *collection_mint_info.key {
+        return Err(MetadataError::NotAMemberOfCollection.into());
     }
 
-    // If the NFT has collection data, we set it to be unverified and then update the collection
-    // size on the Collection Parent.
-    if let Some(collection) = &mut metadata.collection {
-        collection.verified = false;
+    // We need to ensure the metadata is derived from the mint so
+    // someone cannot pass in a burned metadata account associated with
+    // a different mint.
+    assert_metadata_derivation(program_id, collection_metadata_info, collection_mint_info)?;
```

Comment:
Edition account is not validated against the mint but it's also unused in both these handlers.

Reply:
>assert collection verify is valid Does two things:

`assert collection verify is valid` has a third check which it ensures `collection_metadata.mint == collection_mint.key`.

 Since you are no longer calling `assert_collection_verify_is_valid` and now do one of the checks on line 55,  the third check is no longer done, nor is there the edition derivation.

My point in the comment was that both of those checks are not required because they are validations of the collection NFT itself.

And the checks you do have make sense to cover the NFT belongs to the collection and matches the collection mint.